lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -301,20 +389,119 @@ actor ConnectorSink is Sink
 
       // If we have no inputs, then we are not involved in checkpointing.
       if _inputs.size() == 0 then
-        _barrier_initiator.unregister_sink(this)
+        _barrier_coordinator.unregister_sink(this)
         _event_log.unregister_resilient(_sink_id, this)
       end
     end
 
   be report_status(code: ReportStatusCode) =>
     None
 
+  ///////////////
```

Comment:
It seems to me that all of this section's methods could be methods on the same class that encapsulates the 2PC-related fields above.

Reply:
This feels like a Reese's Peanut Butter Cup peanut-butter-in-chocolate is good, but I'll look into pulling them apart.